### PR TITLE
[CELEBORN-548][FLINK] Support flink 1.15.

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
1. Support flink 1.15.
2. Merge common codes.
3. Refine code style.

### Why are the changes needed?
Ditto.

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
UT and cluster.

### CODE REVIEW

File: client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleInputGateDelegation.java [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+        (RemoteShuffleDescriptor) gateDescriptor.getShuffleDescriptors()[0];
+    this.shuffleClient =
+        FlinkShuffleClientImpl.get(
+            remoteShuffleDescriptor.getShuffleResource().getRssMetaServiceHost(),
+            remoteShuffleDescriptor.getShuffleResource().getRssMetaServicePort(),
+            celebornConf,
+            new UserIdentifier("default", "default"));
+
+    this.startSubIndex = startSubIndex;
+    this.endSubIndex = endSubIndex;
```
Comment (zhongqiangczq) [inline client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleInputGateDelegation.java:159]: flink 1.15 should support adaptive scheduler.
`
int startSubIdx = gateDescriptor.getConsumedSubpartitionIndexRange().getStartIndex();
int endSubIdx = gateDescriptor.getConsumedSubpartitionIndexRange().getEndIndex();
`

File: client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/AbstractRemoteShuffleInputGateFactory.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+  protected CelebornConf celebornConf;
+
+  public AbstractRemoteShuffleInputGateFactory(
+      Configuration flinkConf,
+      CelebornConf conf,
+      NetworkBufferPool networkBufferPool,
+      int networkBufferSize) {
+    this.celebornConf = conf;
+    long configuredMemorySize =
```
[the remark this answers is not in the document]
Comment (FMX) [reply client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/AbstractRemoteShuffleInputGateFactory.java:73]: Updated.

**Review by RexXiong — APPROVED** [review]
LGTM, thanks!

**Comment by FMX** [comment]
I tested this patch on flink 1.14 cluster and flink 1.15 cluster. Tests are successful.

**Pull request merged**